### ISSUE TITLE
Concatenate Paramspec False Negatives

### ISSUE THREAD

**Issue by hmc-cs-mdrissi** [microsoft/pyright#3324]
**Describe the bug**
The code below has a buggy decorator that does not pass *args/**kwargs appropriately. It does not give any type errors though. 

```python
from typing import Callable

from typing_extensions import Concatenate, ParamSpec

P = ParamSpec("P")


def dec(func: Callable[Concatenate[int, P], int]) -> Callable[Concatenate[int, P], int]:
    # No Error reported here.
    def _func(x: int) -> int:
        return func(x)  # Expect 1 error here for bad call

    return _func # Expect 2nd error here as _func does not contain P in its signature


def add_k(x: int, k: int) -> int:
    return x + k


dec(add_k)(1, 3) # Crashes
```

The issue partly requires Concatenate. The first missing error does appear if you remove concatenate,

```python

def dec(func: Callable[P, int]) -> Callable[P, int]:
    def _func(x: int) -> int:
        return func(x) # Good error reported

    return _func # Weird, no error here
```

but you can simplify more and have no errors reported even though it'll still crash if you try to use it,

```python

def dec(func: Callable[P, int]) -> Callable[P, int]:
    def _func(x: int) -> int:
        return 3

    return _func # No error
```

So I think there's 2 related issues here.

**Expected behavior**
Errors to be reported when decorator doesn't forward arguments.

**VS Code extension or command-line**
pyright 1.1.235

**Comment by erictraut** [comment]
Thanks for the bug report. As you suspected, there were two separate bugs here. I also found a third one while fixing these two. ParamSpecs add a huge amount of complexity to type checker, so I guess I shouldn't be surprised that we're still finding bugs like this more than a year after ParamSpec was added to pyright.

**Comment by erictraut** [comment]
This is fixed in pyright 1.1.236, which I just published. It will also be included in the next release of pylance.
